Mirroring fails on `anvildev`

And probably also on `dev`, but we aren't sure. The calculated partition size is too large:

```
[ERROR] AssertionError: 33750
Traceback (most recent call last):
  File "/var/task/azul/chalice.py", line 194, in patched_event_source_handler
    return old_handler(self_, event, context)
  File "/var/task/chalice/app.py", line 1761, in __call__
    return self.handler(event_obj)
  File "/var/task/azul/indexer/mirror_controller.py", line 63, in mirror
    self.mirror(event)
  File "/var/task/azul/indexer/mirror_controller.py", line 69, in mirror
    self._handle_events(event, self.service.mirror)
  File "/var/task/azul/indexer/action_controller.py", line 57, in _handle_events
    message_handler(action)
  File "/var/task/azul/indexer/mirror_service.py", line 211, in mirror
    self._queue_messages(self._mirror(action))
  File "/var/task/azul/indexer/mirror_service.py", line 184, in _queue_messages
    return self._queues.send_messages(self._mirror_queue(),
  File "/var/task/azul/queues.py", line 201, in send_messages
    for batch in chunked(messages, self.batch_size):
  File "/opt/python/more_itertools/recipes.py", line 133, in take
    return list(islice(iterable, n))
  File "/var/task/azul/indexer/mirror_service.py", line 233, in _
    partitioned_source = plugin.partition_source_for_mirroring(a.catalog,
  File "/var/task/azul/plugins/__init__.py", line 713, in partition_source_for_mirroring
    return self._partition_source(catalog, source, self.count_files, partition_size)
  File "/var/task/azul/plugins/__init__.py", line 733, in _partition_source
    assert 512 <= partition_size <= 8192, partition_size
```

The solution is to add a call to `max` where the partition size is calculated to ensure it does exceed the upper bound. To facilitate this change, the upper and lower bounds for the assertion should be extracted to class attributes.

--- a/src/azul/indexer/mirror_service.py
+++ b/src/azul/indexer/mirror_service.py
@@ -218,18 +218,18 @@ class MirrorService(BaseMirrorService):
     def _(self, a: MirrorSourceAction) -> Iterable[MirrorAction]:
         assert a.source.id in self._list_public_source_ids(a.catalog), R(
             'Cannot mirror non-public source', a.source)
+        plugin = self._repository_plugin(a.catalog)
         # The desired partition size depends on the maximum number of messages
         # we can send in one Lambda invocation, because queueing the individual
         # mirror_file messages turns out to dominate the running time of
         # handling a mirror_source message.
-        partition_size = int(
+        partition_size = min(plugin.max_partition_size, int(
             aws.sqs_fifo_rate_limit  # max. # of SendMessage calls per second
             * Queues.batch_size  # number of messages per call
             * config.mirror_lambda_timeout  # max. duration of the invocation
             / config.mirroring_concurrency  # number of concurrent invocations
             / 2  # safety margin
-        )
-        plugin = self._repository_plugin(a.catalog)
+        ))
         partitioned_source = plugin.partition_source_for_mirroring(a.catalog,
                                                                    a.source,
                                                                    partition_size)

--- a/src/azul/plugins/__init__.py
+++ b/src/azul/plugins/__init__.py
@@ -733,7 +733,8 @@ class RepositoryPlugin[BUNDLE: Bundle = Bundle[SourcedBundleFQID],
             # We use the "lesser" heuristic during IT to keep the cost and
             # performance of the tests within reasonable limits
             if is_main and not is_it:
-                assert self.min_partition_size <= partition_size <= self.max_partition_size, partition_size
+                assert partition_size >= self.min_partition_size, partition_size
+                assert partition_size <= self.max_partition_size, partition_size
                 prefix = Prefix.for_main_deployment(count, partition_size)
             else:
                 prefix = Prefix.for_lesser_deployment(count)

--- a/src/azul/plugins/__init__.py
+++ b/src/azul/plugins/__init__.py
@@ -688,6 +688,14 @@ class RepositoryPlugin[BUNDLE: Bundle = Bundle[SourcedBundleFQID],
         """
         raise NotImplementedError
 
+    # Sanity-check the partition size. We know the upper bound
+    # caused some mirror Lambda invocations to time out. The lower
+    # bound is hypothetical. It'll likely still work for mirroring,
+    # but we'd like to know if partitions get that small. For
+    # indexing, the partition size is fixed at the upper bound.
+    min_partition_size: ClassVar[int] = 512
+    max_partition_size: ClassVar[int] = 8192
+
     def partition_source_for_indexing(self,
                                       catalog: CatalogName,
                                       source: SOURCE_REF
@@ -697,7 +705,7 @@ class RepositoryPlugin[BUNDLE: Bundle = Bundle[SourcedBundleFQID],
         an updated copy of the source with a heuristically computed prefix that
         should be appropriate for indexing in the given catalog.
         """
-        partition_size = 8192
+        partition_size = self.max_partition_size
         return self._partition_source(catalog, source, self.count_bundles, partition_size)
 
     def partition_source_for_mirroring(self,
@@ -725,12 +733,7 @@ class RepositoryPlugin[BUNDLE: Bundle = Bundle[SourcedBundleFQID],
             # We use the "lesser" heuristic during IT to keep the cost and
             # performance of the tests within reasonable limits
             if is_main and not is_it:
-                # Sanity-check the partition size. We know the upper bound
-                # caused some mirror Lambda invocations to time out. The lower
-                # bound is hypothetical. It'll likely still work for mirroring
-                # but we'd like to know if partitions get that small. For
-                # indexing, the partition size is fixed at the upper bound.
-                assert 512 <= partition_size <= 8192, partition_size
+                assert self.min_partition_size <= partition_size <= self.max_partition_size, partition_size
                 prefix = Prefix.for_main_deployment(count, partition_size)
             else:
                 prefix = Prefix.for_lesser_deployment(count)

